### ISSUE TITLE
32 channels g.tec APP

### ISSUE THREAD

**Issue by cboulay** [labstreaminglayer/App-g.Tec#6]
_From @JARS29 on April 19, 2017 22:13_

Hi!

I'm trying to collect EEG data using g.tec amp 32 channels through LSL. I opened the APP for g.tec, where  16 channels are setup up for default.  When I try to modify the number of channels (including the channel labels) it is not possible (the GUI object for that "number of channels" is blocked ). See the picture:

![image](https://cloud.githubusercontent.com/assets/23078352/25204222/b984c7b2-255c-11e7-89b3-a975b81cdc6f.png)

I tried to modify the file "gUSBamp_config.cfg" adding the channel labels and the number of channels, but when i run the app appear the labels but the object remains blocked with the number 16.

can anyone tell me if with the current LSL APP is possible recording 32 channels?

Now, I'm making a script for collecting the data from g.tec and push it in the LSL. (an update of it coming soon). 

Really, i will appreciate a lot your help, I don't want spend more time on it.  

Thank you.

_Copied from original issue: sccn/lsl_archived#192_

**Comment by cboulay** [comment]
_From @dmedine on April 20, 2017 5:0_

I will have to check the code, but it appears that the app can only 
support 16 channels.

**Comment by cboulay** [comment]
I've created a new g.Tec/g.NEEDaccess application that can now support multiple g.USBamps in master/slave. There are some reasons why one might want to use the direct API instead of the server/client g.NEEDaccess API, so I don't consider this issue solved, but at least there is an alternative for now.

**Comment by cboulay** [comment]
_From @tstenner on October 29, 2018 14:59_

@cboulay Is your gUSB.amp app ready so I can close this?

**Comment by cboulay** [comment]
Nope! There's still a buffer problem at high sampling rates. I haven't had time to get back to it recently. Feel free to move the issue to labstreaminglayer/App-g.Tec

**Comment by windrise** [comment]
Hello, I don't know how to compile the APP for g.tec, Do you have the compiled app for the g.tec client? Could you do me a favor?

**Comment by cboulay** [comment]
It's on the release page: https://github.com/labstreaminglayer/App-g.Tec/releases
It requires the g.NEEDaccess SDK from g.Tec.

If you don't have g.NEEDaccess but you have a g.USBamp and the _very_ old drivers for that then you can will have to compile the gUSB app.

If you have any further questions then please open a new issue.

**Comment by windrise** [comment]
Thank you very very much! 
It's kind of you. Okay

**Comment by kimgemmma** [comment]
Hi,
I have a similar issue, so I thought I would post about it here. I have been trying to record from two g.USBamps simultaneously using the old drivers, as I don't have the g.NEEDaccess SDK. I started the USBamp app twice and connected the amplifiers separately. I kept the sampling frequency at a maximum of 1200 Hz, as I experienced issues when using one amplifier at a higher frequency. However, when recording through LSL, the data for one amplifier does not get stored. This happens randomly. When I start the next recording, nothing gets stored at all and I have to restart everything. Is this a known issue, and is there a solution apart from upgrading to g.NEEDaccess? Any help is appreciated
